### PR TITLE
900 zd wavelength template

### PR DESCRIPTION
**Pull request by bpholden**
Attempt number 2 - only four files are changed. 
Includes:
-  new template
- directions for making new template
- generic howto for making new templates
- changes to spectrograph code to use new template

### CODE REVIEW

**Review by profxj — APPROVED** [review]
Thanks Brad!

Please add a short, one-liner to `CHANGES.rst`

And maybe ask @badpandabear for help on 
adding a new DevSuite test.  Thanks!

**Comment by profxj** [comment]
Also add as a supported grating on front page and `doc/index.rst`

File: doc/construct_template.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+============
+
+To build a new template, the requirements are wavelength calibrations for the
+arc spectra over the full wavelength range needed. This may take multiple
+exposures of arc lamps to cover the full range.
+
+Using the Holy Grail Algorithm
+==============================
+
+Once a set of different arc lamp exposures that cover the desired range,
```
Comment (deborape) [inline doc/construct_template.rst:27]: Once **you have** a set of different arc lamp exposures that cover the desired range

File: doc/construct_template.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+run through the data using the option of the Holy Grail to generate a
+master wavelength calibration, see :doc:`master_wvcalib` for example
```
Comment (deborape) [inline doc/construct_template.rst:29]: master wavelength calibration **and select the slit that have good wavelength calibration by inspecting the QA plots.** See :doc:`master_wvcalib` for example ..

**Review by deborape — APPROVED** [review]
Good job, Brad! I have some comments, but I approve!

Let me know if you want me to run some tests.

**Comment by deborape** [comment]
@bpholden, @profxj, @jhennawi I think I resolved the problem with the blue wavelength calibration for the DEIMOS 900ZD. The calibration was failing because the code was finding a completely wrong shift between the arc frame and the template. The reason, I discovered, is that the code adds padding (i.e., zeros) to the arc spectrum to match the dimension of the template, but it also need to subtract the continuum and determine a significance threshold to identify the lines. This is not really a problem for arcs with a lot of lines, but for blue arcs, the padding is actually dominating the length of the spectrum and heavily affecting the continuum and threshold determination. And this was messing up the X-correlation to find the shift.
I made small changes in `pypeit/core/arc.py` and `pypeit/core/wavecal/autoid.py` to fix this. I tested it with 2 datasets of 900ZD with tilt at 6000Ang and one dataset with tilt at 5500Ang, and it worked great.

I also set as default for DEIMOS that the arcs frames do not have a sigma clipping, i.e., `par['calibrations']['arcframe']['process']['clip'] = False`, because  it generally makes bad MasterArcs if you have arc frames with different lines and the calibration will be failing. I am not sure if it should be the default for all the instruments.

I run the Dev Suite and passed
```
--- PYPEIT DEVELOPMENT SUITE PASSED 94/94 TESTS (Masters ignored) ---
Testing Started at 2021-01-17T18:56:41.337351
Testing Completed at 2021-01-18T01:00:19.963210
Total Time: 6:03:38.625859
```
and checked the wavelength calibration for some of the instruments and looks good.

The unit tests passed too.
`=============== 254 passed in 1058.45s (0:17:38) ====================`

I have a small PR in the Dev Suite to add the files used to create the 900ZD template. see it here:
https://github.com/pypeit/PypeIt-development-suite/pull/138

**Comment by profxj** [comment]
Awesome!!  

@bpholden -- please address the few additional items requested by @deborape 
and then merge if you are happy.  Thanks.

**Pull request merged**